src/Webservices/Profile.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Profile class.
+ *
+ * Profiles are like schemas and define a kind of template which is used to construct a resource
+ * object in memory.  Representations can then be derived from the resource object.  A profile
+ * defines the name and type of each property and could be extended with other information.
+ * An automatic process could be written to generate machine-readable profile documents, such as
+ * ALPS or JSON-LD contexts from the profile.
+ *
+ * New data types can be added by adding a class file in the Type directory.  The new type can
+ * then be referenced in the profile.
+ */
+
+namespace Joomla\Webservices\Webservices;
+
+class Profile
```

Comment:
This class seems pretty broken. The properties are not being set (and getting these are 2/3 of your public methods). The name property which you are setting isn't defined in the class). And your final public method you're debating refactoring out :P I can see your comment at the top trying to explain the intention - but I think I need a bit more explaining as to whether this is actually required in any way